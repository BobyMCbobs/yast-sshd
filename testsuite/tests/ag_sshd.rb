# encoding: utf-8

# ***************************************************************************
#
# Copyright (c) 2000 - 2012 Novell, Inc.
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of version 2 of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, contact Novell, Inc.
#
# To contact Novell about this file by physical or electronic mail,
# you may find current contact information at www.novell.com
#
# ***************************************************************************
module Yast
  class AgSshdClient < Client
    def main
      @READ = {
        "sshd" => {
          "AcceptEnv"              => [
            "LANG LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES ",
            "LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT ",
            "LC_IDENTIFICATION LC_ALL"
          ],
          "PasswordAuthentication" => ["no"],
          "Port"                   => ["22"],
          "Protocol"               => ["2,1"],
          "Subsystem"              => ["sftp\t/usr/lib64/ssh/sftp-server"],
          "UsePAM"                 => ["yes"],
          "X11Forwarding"          => ["yes"]
        }
      }
      @WRITE = {}
      @EXECUTE = {}

      Yast.include self, "testsuite.rb"

      # TESTSUITE_INIT([READ, WRITE, EXECUTE], nil);

      DUMP("------ Dir -----")
      TEST(lambda { SCR.Dir(path(".sshd")) }, [@READ, @WRITE, @EXECUTE], nil)
      DUMP("")

      DUMP("------ Read -----")
      Builtins.foreach(SCR.Dir(path(".sshd"))) do |param|
        TEST(lambda { SCR.Read(Builtins.add(path(".sshd"), param)) }, [
          @READ,
          @WRITE,
          @EXECUTE
        ], nil)
      end

      nil
    end
  end
end

Yast::AgSshdClient.new.main
