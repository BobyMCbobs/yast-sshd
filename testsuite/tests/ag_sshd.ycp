{
    map READ    = $[
	"sshd" : $[
	    "AcceptEnv":[
		"LANG LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES ",
		"LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT ",
		"LC_IDENTIFICATION LC_ALL"
	    ],
	    "PasswordAuthentication":["no"],
	    "Port":["22"],
	    "Protocol":["2,1"],
	    "Subsystem":["sftp\t/usr/lib64/ssh/sftp-server"],
	    "UsePAM":["yes"],
	    "X11Forwarding":["yes"]
	]
    ];
    map WRITE   = $[];
    map EXECUTE = $[];
    
    include "testsuite.ycp";

    // TESTSUITE_INIT([READ, WRITE, EXECUTE], nil);
    
    DUMP("------ Dir -----");
    TEST (``(SCR::Dir(.sshd)), [READ, WRITE, EXECUTE], nil);
    DUMP("");

    DUMP("------ Read -----");
    foreach (string param, (list <string>) SCR::Dir(.sshd), {
	TEST (``(SCR::Read(add(.sshd, param))), [READ, WRITE, EXECUTE], nil);
    });
}
