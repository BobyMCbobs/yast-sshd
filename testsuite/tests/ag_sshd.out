Dump	------ Dir -----
Dir	.sshd: ["AcceptEnv", "PasswordAuthentication", "Port", "Protocol", "Subsystem", "UsePAM", "X11Forwarding"]
Return	["AcceptEnv", "PasswordAuthentication", "Port", "Protocol", "Subsystem", "UsePAM", "X11Forwarding"]
Dump	
Dump	------ Read -----
Dir	.sshd: ["AcceptEnv", "PasswordAuthentication", "Port", "Protocol", "Subsystem", "UsePAM", "X11Forwarding"]
Read	.sshd."AcceptEnv" ["LANG LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES ", "LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT ", "LC_IDENTIFICATION LC_ALL"]
Return	["LANG LC_CTYPE LC_NUMERIC LC_TIME LC_COLLATE LC_MONETARY LC_MESSAGES ", "LC_PAPER LC_NAME LC_ADDRESS LC_TELEPHONE LC_MEASUREMENT ", "LC_IDENTIFICATION LC_ALL"]
Read	.sshd."PasswordAuthentication" ["no"]
Return	["no"]
Read	.sshd."Port" ["22"]
Return	["22"]
Read	.sshd."Protocol" ["2,1"]
Return	["2,1"]
Read	.sshd."Subsystem" ["sftp\t/usr/lib64/ssh/sftp-server"]
Return	["sftp\t/usr/lib64/ssh/sftp-server"]
Read	.sshd."UsePAM" ["yes"]
Return	["yes"]
Read	.sshd."X11Forwarding" ["yes"]
Return	["yes"]
